src/dmd/asttypename.d

```diff
@@ -112,6 +112,7 @@ public :
 unittest
 {
     import dmd.globals : Loc;
-    Expression e = new TypeidExp(Loc.init, null);
+    Loc loc;
+    Expression e = new TypeidExp(loc, null);
```

Comment:
This file isn't part of the DMD build and I only accidentally ended up testing it.
It still doesn't hurt to fix the build failure though.

Reply:
https://github.com/dlang/dmd/blob/67963b95d021e99577b99982d9b1ebd79834ff9a/src/dmd/expression.d#L4758-L4764

That's why `Loc.init` doesn't work. It might be a good idea to use `auto ref` in the constructors.

Reply:
`Loc.initial` is used instead.

Reply:
Thanks!
BTW this looks like a half-finished story -> https://github.com/dlang/dmd/pull/8427